### PR TITLE
Add the Frechet cdf

### PR DESCRIPTION
**Pull request by npfp**
Implement the Fréchet cdf (https://en.wikipedia.org/wiki/Fréchet_distribution).

### CODE REVIEW

**Comment by npfp** [comment]
Purpose is to add the Frechet distribution eventually. But the discussion on PR#843 made me think than maybe I could go with a TransformedDistribution using the property.

If X∼U(0,1) (Uniform distribution (continuous)) then m+s(−log⁡(X))^(−1/α) ∼ Frechet
https://en.wikipedia.org/wiki/Fréchet_distribution

File: tensorflow_probability/python/bijectors/frechet_cdf.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    return self._concentration
+
+  @classmethod
+  def _is_increasing(cls):
+    return True
+
+  def _forward(self, x):
+    with tf.control_dependencies(self._maybe_assert_valid_x(x)):
+      z = (x - self.loc) / self.scale
+      return tf.exp(-z ** (- self.concentration))
```
Comment (srvasude) [inline tensorflow_probability/python/bijectors/frechet_cdf.py]: We generally prefer to tf.convert_to_tensor(...) quantities that are used multiple times before using them below. The test I mention will catch this.

**Comment by npfp** [comment]
I added the Fréchet constraint. Please let me know if there is anything wrong.
Thx

**Comment by npfp** [comment]
Hi @brianwa84, thx!

Looking forward to having it merged. Please let me know if I can be of any help.

Also I'm not sure about the tests that are failing, are they related to my PR?

**Comment by brianwa84** [comment]
While pulling this in I've added a TODO to test a few more extreme values. If you have a moment once this merges, please send a PR to do so and drop the TODO.

**Comment by npfp** [comment]
I'll do. Thx a lot

**Pull request merged**